### PR TITLE
Add support for openai o1 models

### PR DESCRIPTION
**Pull request by Ashad-h**
- Since o1 models doesn't support streaming, I simulated a stream, as it's much easier because all the latitude pipeline seems to be stream only.
- Adapted the playground to display correct loading animation.

### CODE REVIEW

**Comment by geclos** [comment]
Run pnpm prettier to fix any formatting issues

File: packages/core/src/services/ai/index.ts [hunk 1]
```diff
@@ -90,6 +92,67 @@ export async function ai({
...
+    return Result.ok({
+      type: 'text',
+      data: {
+        fullStream: fullStream as any,
+        text: Promise.resolve(result.text),
+        usage: Promise.resolve(result.usage),
+        toolCalls: Promise.resolve(toolCalls),
+      },
+    })
+  }
```
Comment (geclos) [inline packages/core/src/services/ai/index.ts:154]: If you could add some unit tests for this in the accompanying test file it would be awesome 🙏🏼

**Review by geclos — COMMENTED** [review]
fantastic initiative

File: packages/core/src/services/ai/index.ts (hunk 1 above)
Comment (Ashad-h) [reply packages/core/src/services/ai/index.ts:181]: Done :)

**Review by geclos — APPROVED** [review]

**Pull request merged**